src/js_results.hpp

```diff
@@ -74,16 +73,16 @@ struct ResultsClass : ClassDefinition<T, realm::Results, CollectionClass<T>> {
 };
 
 template<typename T>
-typename T::Object ResultsClass<T>::create_instance(ContextType ctx, const realm::Results &results, bool live) {
-    auto new_results = new realm::Results(results);
+typename T::Object ResultsClass<T>::create_instance(ContextType ctx, realm::Results results, bool live) {
+    auto new_results = new realm::Results(std::move(results));
     new_results->set_live(live);
 
     return create_object<T, ResultsClass<T>>(ctx, new_results);
 }
 
 template<typename T>
 typename T::Object ResultsClass<T>::create_instance(ContextType ctx, const realm::List &list, bool live) {
-    return create_instance(ctx, list.get_realm(), list.get_object_schema(), list.get_query(), live);
+    return create_instance(ctx, list.sort({}), live);
```

Comment:
I wasn't sure if was worth adding a new API on `List` just for this, or to add either an implicit or explicit conversion.